components/core/src/compressor_frontend/InputBuffer.cpp

```diff
@@ -0,0 +1,111 @@
+// C++ libraries
+#include <memory.h>
+#include <string>
+
+// spdlog
+#include <spdlog/spdlog.h>
+
+// Project Headers
+#include "InputBuffer.hpp"
+
+using std::string;
+using std::to_string;
+
+namespace compressor_frontend {
+
+    void InputBuffer::reset () {
+        m_at_end_of_file = false;
+        m_finished_reading_file = false;
+        m_consumed_pos = 0;
+        m_bytes_read = 0;
+        m_last_read_first_half = false;
+        Buffer::reset();
```

Comment:
The way you use `Buffer` in `InputBuffer` and `OutputBuffer` suggests to me that `Buffer` would be more appropriate as a class member in the `XBuffer` rather as a parent class. That would also get rid of this kind of antipattern where the overridden method is *required* to call the parent method.